File: src/models/ActivityLog.ts
import type { ActivityEntityType, Role } from "@/types/auth";
import type { Json } from "@/types/supabase";

export type ActivityLogRow = {
  id: string;
  actor_role: Role;
  actor_id: string | null;
  entity_type: ActivityEntityType | string;
  entity_id: string | null;
  action: string;
  description: string | null;
  metadata: Json | null;
  created_at: string;
};

export type ActivityLog = {
  id: string;
  actorRole: Role;
  actorId: string | null;
  entityType: ActivityEntityType | string;
  entityId: string | null;
  action: string;
  description: string | null;
  metadata: Json | null;
  createdAt: string;
};

File: rkispro-android/app/src/main/java/com/rkispro/marketplace/data/model/ActivityLog.kt
package com.rkispro.marketplace.data.model

import com.google.gson.annotations.SerializedName

data class ActivityLog(
    val id: String,
    @SerializedName("actor_role") val actorRole: String,
    @SerializedName("actor_id") val actorId: String,
    @SerializedName("entity_type") val entityType: String,
    @SerializedName("entity_id") val entityId: String,
    val action: String,
    val description: String? = null,
    val metadata: Map<String, Any>? = null,
    @SerializedName("created_at") val createdAt: String? = null
)
